grpc/federation/cel/metadata.go

```diff
@@ -0,0 +1,61 @@
+package cel
+
+import (
+	"context"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"google.golang.org/grpc/metadata"
+)
+
+const MetadataPackageName = "metadata"
+
+func NewMetadataLibrary(ctx context.Context) *MetadataLibrary {
+	return &MetadataLibrary{
+		ctx: ctx,
+	}
+}
+
+type MetadataLibrary struct {
+	ctx context.Context
+}
+
+func (lib *MetadataLibrary) LibraryName() string {
+	return packageName(MetadataPackageName)
+}
+
+func createMetadata(name string) string {
+	return createName(MetadataPackageName, name)
+}
+
+func createMetadataID(name string) string {
+	return createID(MetadataPackageName, name)
+}
+
+func (lib *MetadataLibrary) CompileOptions() []cel.EnvOption {
+	opts := []cel.EnvOption{
+		cel.Function(
+			createMetadata("INCOMING"),
```

Comment:
I’d like to use UPPER CASE name for constant value. So in this case, I think better to use lower case name ( `incoming` ).